File: ePOS2EasySelect_ParseQR/ePOS2EasySelect_ParseQR/IndicatorView.m
#import "IndicatorView.h"

@interface IndicatorView () {
    UIActivityIndicatorView *indicator_;
}
@end

@implementation IndicatorView

- (void)dealloc
{
    [self hide];
}

- (void)show:(UIView *)base
{
    if (!base) return;

    if (indicator_) return;

    self.backgroundColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:0.5];
    self.frame = CGRectMake(0, 0, MAX(base.bounds.size.width, base.bounds.size.height),
                            MAX(base.bounds.size.width, base.bounds.size.height));
    self.hidden = NO;
    [base addSubview:self];

    indicator_ =
        [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleWhiteLarge];
    indicator_.bounds = CGRectMake(0, 0, 36, 36);
    indicator_.center = base.center;
    [self addSubview:indicator_];

    [indicator_ startAnimating];
}

- (void)hide
{
    if (!indicator_) {
        return;
    }

    [indicator_ stopAnimating];
    [self removeFromSuperview];
    indicator_ = nil;
}

@end

File: ePOS2EasySelect_ParseNFC/ePOS2EasySelect_ParseNFC/IndicatorView.swift
import UIKit

class IndicatorView: UIView {
    
    // MARK: - Properties
    
    private var indicator: UIActivityIndicatorView?
    
    // MARK: - Method
    
    deinit {
        hide()
    }
    
    func show(baseView: UIView?) {
        guard let baseView = baseView, self.indicator == nil else {
            return
        }
        
        self.backgroundColor = UIColor.black.withAlphaComponent(0.5)
        self.frame = CGRect(x: 0, y: 0, width: baseView.bounds.size.width, height: baseView.bounds.size.height)
        self.isHidden = false
        baseView.addSubview(self)
        
        let indicator = UIActivityIndicatorView(activityIndicatorStyle: .whiteLarge)
        self.indicator = indicator
        indicator.bounds = CGRect(x: 0, y: 0, width: 36, height: 36)
        indicator.center = baseView.center
        indicator.hidesWhenStopped = true
        self.addSubview(indicator)
        
        indicator.startAnimating()
    }
    
    func hide() {
        guard let indcator = self.indicator else {
            return
        }
        
        indcator.stopAnimating()
        self.removeFromSuperview()
        self.indicator = nil
    }
}
